src/controllers/Interactions.ts

```diff
@@ -46,18 +47,38 @@ export async function handleInteraction(buttonInteraction: ButtonInteraction): P
     }
 
     case ButtonCustomID.CreateRandomTeam: {
-      const currentMatch = await createRandomMatch();
-      const emptyQueue: PlayerInQueue[] = [];
+      const playerInQueue = await QueueRepository.isPlayerInQueue(buttonInteraction.user.id);
+
+      if (!playerInQueue) {
+        break;
+      } else {
+        const currentMatch = await createRandomMatch();
+        const emptyQueue: PlayerInQueue[] = [];
 
-      await Promise.all([
-        //Create new reply to start a match
-        await message.channel.send(MessageBuilder.activeMatchMessage(currentMatch)),
+        await Promise.all([
+          //Create new reply to start a match
+          await message.channel.send(MessageBuilder.activeMatchMessage(currentMatch)),
 
-        //Update the embed with an empty queue message
-        await message.edit(MessageBuilder.queueMessage(emptyQueue)),
-      ]);
+          //Update the embed with an empty queue message
+          await message.edit(MessageBuilder.queueMessage(emptyQueue)),
+        ]);
 
-      break;
+        break;
+      }
+    }
+
+    case ButtonCustomID.ChooseTeam: {
+      const playerInQueue = await QueueRepository.isPlayerInQueue(buttonInteraction.user.id);
+
+      if (!playerInQueue) {
```

Comment:
Small thing here, you can rewrite this like:
```
if (playerInQueue) {
   // do queue stuff
}
```
and then you don't need an else. Just reads a bit better.

Reply:
Fixed in next commit